tests/src/testcommon.hh

```diff
@@ -187,139 +186,28 @@ template <typename NonLinearOperator>
-[[nodiscard]] auto checkLinearStressOf2DElement(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
-                                                const std::string& messageIfFailed = "") {
-  static_assert(FiniteElement::Traits::mydim == 2,
-                "The test to check linear stress is only supported for the two-dimensional case");
-  assert(fe.localView().element().type() == Dune::GeometryTypes::quadrilateral
-         && "The test to check linear stress is only supported for quadrilaterals");
-  static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == 2,
-                "The test to check linear stress is only supported with the powerBasis being two-dimensional");
-  static_assert(std::is_same_v<std::remove_cvref_t<decltype(fe.localView().tree().child(0))>,
-                               Dune::Functions::LagrangeNode<
-                                   std::remove_cvref_t<decltype(fe.localView().globalBasis().gridView())>, 1, double>>,
-                "The test to check linear stress is only supported for a linear Lagrange basis");
-
-  Dune::TestSuite t("Linear Stress check of 2D solid element (4-node quadrilateral)");
-  using namespace Ikarus;
-  using namespace Dune::Indices;
-  using namespace Dune::Functions::BasisFactory;
-  constexpr int gridDim = 2;
-
-  auto& displacement = nonLinearOperator.firstParameter();
-  displacement << 0, 0, 1, 1, 1, 1, 1, 1;
-  std::array<Dune::FieldVector<double, 3>, 4> expectedStress = {{{1428.5714287000, 1428.5714287000, 769.2307693000},
-                                                                 {1098.9010990000, 329.6703297000, 384.6153846500},
-                                                                 {329.6703297000, 1098.9010990000, 384.6153846500},
-                                                                 {0, 0, 0}}};
-
-  /// Modifications for the EAS method
-  /// The test is only to check the computation of Cauchy stress in the linear elastic case. This includes only the
-  /// addition of C*M*alpha to the stress from pure displacement formulation. Hence it is sufficient to test only for
-  /// one of the EAS variant as the other M matrices are already tested.
-  if constexpr (requires { fe.setEASType(4); }) {
-    fe.setEASType(4);
-    expectedStress = {{{1214.2857143755, 1214.2857143755, 384.6153846800},
-                       {1214.2857144055, 214.2857142945, 384.6153847400},
-                       {214.2857142945, 1214.2857144055, 384.6153845600},
-                       {214.2857143245, 214.2857143245, 384.6153846200}}};
+[[nodiscard]] auto checkCalculateAt(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
+                                    const Ikarus::ResultRequirements<>& resultRequirements, const auto& expectedResult,
+                                    const auto& evaluationPositions, const std::string& messageIfFailed = "") {
+  Eigen::MatrixXd computedResults(expectedResult.rows(), expectedResult.cols());
```

Comment:
You can also write here:

```cpp
Eigen::MatrixXd computedResults;
computedResults.setZero(expectedResult.rows(), expectedResult.cols());
```

or 

```cpp
Eigen::MatrixXd computedResults= Eigen::MatrixXd::Zero(expectedResult.rows(), expectedResult.cols());
```

or not setting it zero any since it will be overwritten anyway. Setting the size will be enough.
```cpp
Eigen::MatrixXd computedResults(expectedResult.rows(), expectedResult.cols());
```